File: packages/shared/src/verdicts.ts
/**
 * Veredictos canónicos del juez (plan v2.0 §5.3).
 *
 * Espejo exacto de `judge/verdicts.py`. La correspondencia se verifica en
 * `tests/limits-contract.test.ts`; si divergen, CI falla.
 */
export const VERDICTS = {
  AC: 'AC',
  WA: 'WA',
  TLE: 'TLE',
  MLE: 'MLE',
  RE: 'RE',
  CE: 'CE',
  OLE: 'OLE',
  SE: 'SE',
} as const;

export type Verdict = (typeof VERDICTS)[keyof typeof VERDICTS];

/** Etiquetas en español para mostrar al jugador. */
export const VERDICT_LABELS: Record<Verdict, string> = {
  AC: 'Aceptado',
  WA: 'Respuesta incorrecta',
  TLE: 'Tiempo excedido',
  MLE: 'Memoria excedida',
  RE: 'Error en ejecución',
  CE: 'Error de compilación',
  OLE: 'Salida excedida',
  SE: 'Error del sistema',
};

export function isVerdict(value: unknown): value is Verdict {
  return typeof value === 'string' && value in VERDICTS;
}

File: judge/verdicts.py
"""Veredictos canónicos del juez (plan v2.0 §5.3).

Espejo exacto de `packages/shared/src/verdicts.ts`. La correspondencia se
verifica en `packages/shared/tests/limits-contract.test.ts`.
"""

from enum import StrEnum


class Verdict(StrEnum):
    AC = "AC"    # Todos los casos pasan
    WA = "WA"    # Salida difiere en >=1 caso
    TLE = "TLE"  # Excede tiempo en >=1 caso
    MLE = "MLE"  # OOM-kill del kernel
    RE = "RE"    # Exit code != 0
    CE = "CE"    # Falla compilación
    OLE = "OLE"  # stdout > 1 MB
    SE = "SE"    # Fallo interno del juez (se reintenta 1 vez)
